criar.sql

```diff
@@ -21,141 +21,141 @@ DROP TABLE IF EXISTS CheckIn;
 DROP TABLE IF EXISTS Status; --NOVO! ADICIONAR NOS PONTOS ANTERIORES
 
 CREATE TABLE Company (
-	CID		INTEGER,
-	Cname 		TEXT,
-	NIF		INTEGER,
-	address		TEXT,
+	CID			INTEGER,
+	Cname 		TEXT NOT NULL,
+	NIF			INTEGER UNIQUE NOT NULL,
+	address		TEXT NOT NULL,
 	CONSTRAINT company_pk PRIMARY KEY (CID)
 );
 CREATE TABLE Document (
 	docID		INTEGER PRIMARY KEY,
 	dNAME		TEXT NOT NULL,
-	date_ini	DATE,
+	date_ini	DATE NOT NULL,
 	date_fin	DATE,
 	type		TEXT,
 	ownerID		INTEGER,
-	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID),
-	CONSTRAINT document_owner_fk FOREIGN KEY (ownerID) REFERENCES Worker(WID),
+	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID) ON DELETE CASCADE,

```

Comment:
O Type nunca será apagado, apenas fica com o status diferente. Não deve haver problema deixar isto assim

Reply:
É melhor ficar NULL do que apagar o Documento. Em princípio o type não se apaga, certo, mas acontecendo um erro qualquer é preferível que se altere só para NULL do que apagar o documento todo.

Reply:
Alternativamente usar RESTRICT para não deixar apagar tipos que sejam usados nos documentos, tb é uma boa solução

Reply:
```suggestion
	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID) ON DELETE SET NULL,
```
Só testando coisas, é possivel dar commit da sugestions

Reply:
```suggestion
	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID) ON DELETE RESTRICT,
```
Acho que assim fica ainda melhor do que SET NULL

Reply:
Fica on delete restrict no type e no status então